website/content/specification/syntax/constraints.md

```diff
@@ -164,6 +164,21 @@ And the following document.
 The expect constraint would pass for each `sibling` in the `parent` named "p1", and would fail for each `sibling` in the `parent` named "p2".
 
 
+## `is-unique` constraints
+
+The `is-unique` constraint is a type of Metaschema constraint that restricts field or flag value(s) based on a component not being referenced more than once.
+
+The `id` attribute of an `is-unique` constraint specifies an identifier for the constraint.
+
+The `@target` attribute of an `<is-unique>` constraint specifies the node in a component definition whose value is distinct. The `is-unique` constraint MUST define a [`@target`](#target) with a Metapath expression.
+
+The `key-field` child element of an `is-unique` constraint defines the attribute or element value that is the key or uuid. **???** The `key-field` element MUST define a [`@target`](#target) attribute with a Metapath expression evaluated relative to [the evaluation focus](#constraint-processing) of the constraint's `@target`.
+
+An `is-unique` constraint MAY define only one `@key-field`. 
```

Comment:
I believe this is a must no? I'll have to check when back from lunch.

```suggestion
An `is-unique` constraint MUST define only one `@key-field`. 
```

Reply:
This is a typo. It should read: **MUST**

Reply:
I actually take this back as I was confused, I need to amend my suggestion as well: did you test this out with the CLI or XSLT tools? I believe the document and functionality should match `index-has-key` or `index` in this regard, other parts are different.

```suggestion
An `is-unique` constraint MAY define more than one `<key-field>` child element. The composite key for each entry in the index is the combination of values for the `@target` of every `<key-field/>`. The composite values of the key are the discriminator for the uniqueness of the index entry.
```